### ISSUE TITLE
`using module "..\PathRelativeModule"` rejected by Intellisense

### ISSUE THREAD

**Issue by AndrewGaspar** [PowerShell/vscode-powershell#64]
I've got a project with a file hierarchy like this:

```
\
  \ Module
    Module.psd1
    Module.psm1
  \ Tests
    Module.Tests.ps1
```

At the top of Module.Tests.ps1, I have

``` powershell
using module "..\Module"
```

PowerShell itself seems to accept this statement (it loads all the module's functions and types as expected), but VS Code rejects it with the error "Could not find the module '..\Module'.". I believe this Intellisense is incorrect, but my understanding of the "using module" syntax is admittedly incomplete.

**Comment by rkeithhill** [comment]
I don't think there is such a construct as `using module` in PowerShell.  In early v5 drops there was some support for `using <namespace>` but that just pulled in namespace names. 

I believe what you want at the top of your test file is:
```powershell
Import-Module ..\Module\Module.psd1
```
Note that quotes aren't needed around path arguments unless the path contains spaces.

**Comment by AndrewGaspar** [comment]
Well, both PowerShell the shell and the VS Code extension accept the syntax. VS Code only has an issue when it can't find the module path, but that only seems to be on this relative path syntax. For example, VS Code has no problem with this fully qualified path:
```powershell
using module ~\Projects\CmdUsageSyntax\CmdUsageSyntax
```

So the feature is at the very least build in to the version of PowerShell I'm using. Though I haven't been able to find any good documentation on it, so it may very well not be supported.

**Comment by AndrewGaspar** [comment]
For reference, PowerShell ISE is able to resolve the path correctly when opening the script. However, if I point it at a path that does not resolve to a module, it raises an error on the using statement, so it is able to correctly distinguish between relative paths that contain a module and those that do not.

**Comment by AndrewGaspar** [comment]
The bug is most likely with https://github.com/PowerShell/PowerShellEditorServices and not with the vscode extension directly.

**Comment by rkeithhill** [comment]
Looks like we got more features with the v5 `using` statement than I was aware of.  So yeah, it appears that `relative` path Intellisense in the extension for VSCode is less than desirable as it is based off of the VSCode install dir.  That's the bug here I think.

I was able to successfully import a module using a relative path:
```powershell
using module ..\..\..\..\Documents\WindowsPowerShell\Modules\Await
```
The kicker is that the relative path is relative to where the current script resides (not the current dir).  

So I would have expected `using module ..\Module` to have actually worked during debug even though the editor was flagging the path as `Could not find module...`.  At least, that is what I see.

**Comment by AndrewGaspar** [comment]
Yeah, for `using module` specifically, the relative path is to *the file* not the cwd. I don't really expect that you fake relative directories generally in VS Code, since by design it's flakey. But specifically with this `using module` pattern, it seems like it should be a supported scenario.

It would be killer if you could auto-fill the values for $PSScriptRoot and $PSCommandPath and possibly resolve `Import-Module` actions that way, and thus provide IntelliSense for those modules. But I can see why that would be needlessly difficult.

**Comment by daviwil** [comment]
Thanks for the report, Andrew!  As Keith identifies, this definitely seems to be an issue with the presumed cwd at the time the module resolve operation time takes place.  We'll get this fixed in PowerShell Editor Services, I've filed a bug here:

https://github.com/PowerShell/PowerShellEditorServices/issues/99

**Comment by tylerl0706** [comment]
This was fixed in PowerShell/PowerShellEditorServices#99! I'm going to go ahead and close this now. Feel free to open a new issue if you're still having trouble.
